File: 2021/day-1/part-1/main.cpp
#include <iostream>
#include <fstream>
#include <vector>

int main(){
    std::ifstream FILE("inputs.txt");
    
    int count = 0;
    int plug = -1;
    
    for (std::string line; std::getline(FILE, line);) {
        int x = std::stoi(line);
        if (plug != -1 && x > plug) {
            count++;
        }
        plug = x;
    }
    std::cout << count;
    return 0;
}

File: 2021/day-1/part-1/main.jl
function main()::Int
    filedata = open("inputs.txt", "r");
    data = readlines(filedata);
    count = 0;
    for (index, line) in enumerate(data)
        if parse(Int, line) > parse(Int, index == 1 ? line : data[index-1])
            count += 1;
        end
    end
    return count
end
println(main())